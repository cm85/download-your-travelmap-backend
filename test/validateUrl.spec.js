const validateUrl = require('../app/validateUrl');

test('happy path', () => {
  const result = validateUrl('http://www.tripadvisor.com/abc');
  expect(result).toMatchSnapshot();
});

test('invalid url', () => {
  function validate() {
    validateUrl('http://www.wtf/abc');
  }
  expect(validate).toThrowErrorMatchingSnapshot();
});


test('no url', () => {
  function validate() {
    validateUrl('x');
  }
  expect(validate).toThrowErrorMatchingSnapshot();
});
