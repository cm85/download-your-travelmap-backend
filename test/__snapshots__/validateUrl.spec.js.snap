// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`happy path 1`] = `true`;

exports[`invalid url 1`] = `"Please enter your TripAdvisor profile URL"`;

exports[`no url 1`] = `"Please enter a valid URL"`;
